graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -357,12 +357,21 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
                                                      inner_location, context)
         basic_blocks.extend(inner_basic_blocks)
 
+    marked_location_stack = context['marked_location_stack']
+
     # step V-3: mark the graph position, and process output_source directive
     if not is_in_fold_scope(context):
         # We only mark the position if we aren't in a folded scope.
         # Folded scopes don't actually traverse to the location, so it's never really visited.
         context['location_types'][location] = strip_non_null_from_type(current_schema_type)
         basic_blocks.append(_mark_location(location))
+        # The marked_location_stack explicitly maintains a stack (implemented as list)
+        # of dicts (each corresponding to a MarkLocation) containing:
+        #  - location: the location within the corresponding MarkLocation object
+        #  - num_traverses: the number of Recurse and Traverse blocks created
+        #                   after the corresponding MarkLocation
+        # The following append is the initial MarkLocation for the current vertex.
```

Comment:
This is misleading, there is no `MarkLocation` block here -- only a `Location` and an int counter.